### DISCUSSION TITLE
Form Select Create New Record Inline By Typing Something That Doesn't Exist

### DISCUSSION THREAD

**Discussion by nickescobedo** [filamentphp/filament#9415] [Help]
Is it possible to create a new option based on if someone types something in a select that doesn't exist without using a modal?

For example, let's say we have a list of car brands Ford and Honda.
Someone goes to search for Toyota, but it doesn't exist.
So on save a new car brand is created and then associated with the form record.

Thanks!

**Comment by sinnbeck** [comment]
As far as I know that would require the input to be a text input `<input type="text"`. And filament uses a `<select`

Filament uses choices.js
https://choices-js.github.io/Choices/

**Reply by sinnbeck** [reply]
Maybe you could use tags instead?

https://filamentphp.com/docs/3.x/forms/fields/tags-input

**Reply by nickescobedo** [reply]
I only want the person to be able to choose one thing, so tags wouldn't work. I'll take a look at choices.js. Thank you for the response.
My guess is that it doesn't exist and if I want it I'll have to make it custom.

**Reply by sinnbeck** [reply]
Yeah I think you can check out the implementation of filament select and just make your own custom version based on choicesjs with a text input instead of a select :)

But it might be worth checking that it allows a max of 1 items first

**Reply by nickescobedo** [reply]
This might work via choices.js.
![image](https://github.com/filamentphp/filament/assets/2837169/71947ae5-e4a6-483f-a6ba-2a731b50e8bc)

**Comment by jmarkfen** [comment]
I think having the closure in `getSearchResultsUsing()` of a SelectInput return an array that contains the `$search` string could work. Then customize the saving process of the Create or Edit page.

**Comment by helturkey** [comment]
I think you need to take a look on this link:
https://filamentphp.com/docs/3.x/forms/fields/select#creating-a-new-option-in-a-modal
